### ISSUE TITLE
1.14.1 Crash when loading feed entries on Wayland muslc linux x86_64

### ISSUE THREAD

**Issue by aelspire** [skyjake/lagrange#562]
Hi, thanks for such great Gemini client!

I noticed that current version crashes on start when run on Wayland and compiled with musl. I'm not sure if it happened before because I've just changed my linux distro.

Lagrange version 1.14.1

```
#0  0x00007ffff7fb6760 in pthread_mutex_lock () from /lib/ld-musl-x86_64.so.1
#1  0x00007ffff7f231d9 in mtx_lock () from /lib/lib_Foundation.so.1.5
#2  0x00007ffff7f15e69 in lock_Mutex () from /lib/lib_Foundation.so.1.5
#3  0x0000555555586db7 in listEntries_Feeds ()
    at /home/aelspire/Projects/lagrange/src/feeds.c:805
#4  numUnread_Feeds () at /home/aelspire/Projects/lagrange/src/feeds.c:821
#5  0x00005555555e3843 in processEvent_SidebarWidget_ (d=0x7fffe5779100, 
    ev=0x7fffffffe170)
    at /home/aelspire/Projects/lagrange/src/ui/sidebarwidget.c:1520
#6  0x0000555555621f8e in dispatchEvent_Widget (d=d@entry=0x7fffe5779100, 
    ev=ev@entry=0x7fffffffe170)
    at /home/aelspire/Projects/lagrange/src/ui/widget.c:1243
#7  0x0000555555621f66 in dispatchEvent_Widget (d=d@entry=0x7fffe5772780, 
    ev=ev@entry=0x7fffffffe170)
    at /home/aelspire/Projects/lagrange/src/ui/widget.c:1205
#8  0x0000555555621f66 in dispatchEvent_Widget (d=d@entry=0x7fffe57724b0, 
    ev=ev@entry=0x7fffffffe170)
    at /home/aelspire/Projects/lagrange/src/ui/widget.c:1205
#9  0x0000555555621f66 in dispatchEvent_Widget (d=d@entry=0x7fffe57723c0, 
    ev=ev@entry=0x7fffffffe170)
    at /home/aelspire/Projects/lagrange/src/ui/widget.c:1205
#10 0x0000555555621f66 in dispatchEvent_Widget (d=d@entry=0x7fffe5b92c40, 
    ev=ev@entry=0x7fffffffe170)
    at /home/aelspire/Projects/lagrange/src/ui/widget.c:1205
#11 0x0000555555621f66 in dispatchEvent_Widget (d=0x7fffe5b92b50, 
    ev=ev@entry=0x7fffffffe170)
    at /home/aelspire/Projects/lagrange/src/ui/widget.c:1205
#12 0x000055555561b17b in dispatchEvent_Window (d=d@entry=0x7fffedd03dc0, 
    ev=ev@entry=0x7fffffffe170)
    at /home/aelspire/Projects/lagrange/src/ui/window.c:1349
#13 0x000055555561c21a in processEvent_Window (d=d@entry=0x7fffedd03dc0, 
    ev=ev@entry=0x7fffffffe2b0)
    at /home/aelspire/Projects/lagrange/src/ui/window.c:1226
#14 0x000055555557fcda in processEvents_App (
    eventMode=eventMode@entry=waitForNewEvents_AppEventMode)
    at /home/aelspire/Projects/lagrange/src/app.c:1873
#15 0x0000555555582277 in run_App_ (d=0x55555568e640 <app_>)
    at /home/aelspire/Projects/lagrange/src/app.c:2055
#16 run_App (argc=argc@entry=1, argv=argv@entry=0x7fffffffe438)
    at /home/aelspire/Projects/lagrange/src/app.c:2209
#17 0x0000555555575d79 in main (argc=1, argv=0x7fffffffe438) at /home/aelspire/Projects/lagrange/src/main.c:87
(gdb) f 1
#1  0x00007ffff7f231d9 in mtx_lock () from /lib/lib_Foundation.so.1.5
(gdb) f 3
#3  0x0000555555586db7 in listEntries_Feeds () at /home/aelspire/Projects/lagrange/src/feeds.c:805
805	   lock_Mutex(d->mtx);
(gdb) l
800	   return -cmp_Time(&(*e1)->discovered, &(*e2)->discovered);
801	}
802	
803	const iPtrArray *listEntries_Feeds(void) {
804	   iFeeds *d = &feeds_;
805	   lock_Mutex(d->mtx);
806	   /* The worker will never delete feed entries so we can use the same ones. Just make a copy
807	      of the array in case the worker modifies it. */
808	   iPtrArray *list = collect_PtrArray(copy_Array(&d->entries.values));
809	   unlock_Mutex(d->mtx);
(gdb) print d
$1 = (iFeeds *) 0x55555568e960 <feeds_>
(gdb) print d->mtx
$2 = (iMutex *) 0x0
```

I've tried debugging this and now I cannot get previous trace, now  I'm getting:

```
Thread 23 "SDLTimer" received signal SIGSEGV, Segmentation fault.
[Switching to LWP 18162]
0x00007ffff7fb75e9 in pthread_setname_np () from /lib/ld-musl-x86_64.so.1
(gdb) bt
#0  0x00007ffff7fb75e9 in pthread_setname_np () from /lib/ld-musl-x86_64.so.1
#1  0x0000000000000000 in ?? ()
```

I'm not sure what debugging symbols I need to get full trace, I'll investigate it a little bit more.

**Comment by skyjake** [comment]
Thanks for the debugging info. Would be nice to see a consistently reproduced trace...

For the record, some time ago I fixed a couple of things to get Lagrange running on Alpine Linux. This was probably in early 2021, though, so it's possible that some new issues have cropped up.

**Comment by skyjake** [comment]
This should be fixed by #565.
